src/core/api/coap_api.cpp

```diff
@@ -66,6 +66,11 @@ otError otCoapHeaderAppendUintOption(otCoapHeader *aHeader, uint16_t aNumber, ui
     return static_cast<Coap::Header *>(aHeader)->AppendUintOption(aNumber, aValue);
 }
 
+otError otCoapHeaderSetContentFormatOption(otCoapHeader *aHeader, otCoapOptionContentFormat aContentFormat)
+{
+    return otCoapHeaderAppendUintOption(aHeader, OT_COAP_OPTION_CONTENT_FORMAT, (uint32_t)aContentFormat);
```

Comment:
Ahh okay, didn't spot the `AppendContentFormatOption` function and its enum… maybe because I was looking at the public API headers which seemed to make no mention of it.  My work here is basically monkey-see monkey do. :-)

I wasn't sure if it should be a "set content format" or "append content format"… guess I'm used to HTTP and while CoAP is similar in some ways, it is different in others.

I'm definitely in favour of re-using existing code, so I'll have a look at this function and see if I can re-work it to use the new enum since that's written with the newer coding style (consistent with what I've seen elsewhere in OpenThread).

That, and I'll poke @patrickc77 to see if I can get that CLA sorted out for @vrtsystems.